pkg/discoveredpolicy/controllers/policy_deployment.go

```diff
@@ -0,0 +1,340 @@
+package controllers
+
+import (
+	"context"
+	"reflect"
+	"time"
+
+	securityv1 "github.com/accuknox/auto-policy-discovery/pkg/discoveredpolicy/api/security.kubearmor.com/v1"
+	ciliumv2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2"
+	kspv1 "github.com/kubearmor/KubeArmor/pkg/KubeArmorController/api/security.kubearmor.com/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/client-go/kubernetes/scheme"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+)
+
+func (r *DiscoveredPolicyReconciler) handleKubeArmorPolicy(ctx context.Context, request ctrl.Request, instance *securityv1.DiscoveredPolicy) (*ctrl.Result, error) {
```

Comment:
`handleKubeArmorPolicy` , `handleK8sNetworkPolicy` and `handleCiliumNetworkPolicy`  have similar code, can we create a common function and send policy type in it

Reply:
@Shreyas220 , this will envolve using generic (not supported for stucts at the moments) or type assertions (unsafe and has an impact on performance).